// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package apps.proto;

// This file represents the format of the deduplication data stored on disk.
// The DeduplicateResponse proto will be mapped to the DeduplicateData proto
// before being saved to disk.

message DeduplicateData {
  // An Application Group representing a set of "duplicate" apps.
  message AppGroup {
    optional string app_group_uuid = 1;

    // Applications which belong to the App Group.
    repeated string package_id = 2;
  }

  // The list of Application Groups corresponding to the requested applications.
  repeated AppGroup app_group = 1;
}
